tests/test_analysis.py

```diff
@@ -0,0 +1,87 @@
+import covasim as cv
+import unittest
+import numpy as np
+
+
+# Suite of tests to test basic functionality of the analysis.py file
+def test_analysis_snapshot():
+    sim = cv.Sim(analyzers=cv.snapshot('2020-04-04', '2020-04-14'))
+    sim.run()
+    snapshot = sim['analyzers'][0]
+    people = snapshot.snapshots[0]             # Option 1
+    people2 = snapshot.snapshots['2020-04-04'] # Option 2
+    people3 = snapshot.get('2020-04-14')       # Option 3
+    people4 = snapshot.get(34)                 # Option 4
+    people5 = snapshot.get()                   # Option 5
+    # people3 = []  # uncomment to verify error
+    i = 0
+    peoples = [people, people2, people3, people4, people5]
+    for p in peoples:
+        i+=1
+        if len(p) == 0:
+            raise ValueError("Option {} is not getting people array correctly".format(i))
+
+
+def test_analysis_hist():
+    # raising multiple histograms to check windows functionality
+    sim = cv.Sim(analyzers=cv.age_histogram(days=["2020-03-30", "2020-03-31", "2020-04-01"]))
+    sim.run()
+
+    # checks to make sure dictionary form has right keys
+    agehistDict = sim['analyzers'][0].get()
+    assert len(agehistDict.keys()) == 5
+
+    # checks to see that compute windows is correct
+    agehist = sim['analyzers'][0]
+    try:
+        agehist.compute_windows()
+    except ValueError:
+        raise ValueError("Unable to compute windows")
+
+    # checks compute_windows and plot()
+    try:
+        agehist.plot(windows=True)

```

Comment:
Is there a way to save this to disk rather than make it appear?  Then we could assert for the existence of the file (make sure it isn't there, generate the plot, make sure it is there, delete the new file)
```
expected_file = "my_age_hist_plot.png"
assert not os.path.isfile(expected_file), msg=f"File {expected_file} shouldn't be here yet. Please clean up."
agehist.save_the_thing(filename=expected_file)
assert os.path.isfile(expected_file), msg=f"File {expected_file} should now be written to disk, was not found."
os.unlink(expected_file) # remove file and don't leave garbage behind. :)
```

Reply:
In this case I think it's OK to not write anything to file. It takes a relatively large proportion of CPU time and I don't think it would tell us much (it could fail to write the file due to issues with the Matplotlib backend or file permissions that don't impact the Covasim code, or it could succeed to write the file but it could be blank or otherwise not the right content).